bundles/org.openhab.core.addon.marketplace/src/main/java/org/openhab/core/addon/marketplace/internal/json/JsonMarketplaceAddonService.java

```diff
@@ -0,0 +1,249 @@
+
+    @Activate
+    public void activate(Map<String, Object> config) {
+        modified(config);
+    }
+
+    @Modified
+    public void modified(Map<String, Object> config) {
+        String urls = Objects.requireNonNullElse((String) config.get("urls"), "");
+        marketplaceUrls = Arrays.asList(urls.split(","));
+        showUnstable = (Boolean) config.getOrDefault("showUnstable", false);
+        refreshSource();
+    }
+
+    @Reference(cardinality = ReferenceCardinality.AT_LEAST_ONE, policy = ReferencePolicy.DYNAMIC)
+    protected void addAddonHandler(MarketplaceAddonHandler handler) {
+        this.addonHandlers.add(handler);
+    }
+
+    protected void removeAddonHandler(MarketplaceAddonHandler handler) {
+        this.addonHandlers.remove(handler);
+    }
+
+    @Reference
+    protected void setEventPublisher(EventPublisher eventPublisher) {
+        this.eventPublisher = eventPublisher;
+    }
+
+    protected void unsetEventPublisher(EventPublisher eventPublisher) {
+        this.eventPublisher = null;
+    }
+
+    @Override
+    public String getId() {
+        return "jsonmarketplace";
+    }
+
+    @Override
+    public String getName() {
+        return "JSON 3rd Party Marketplace";
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void refreshSource() {
+        cachedAddons = (List<AddonEntryDTO>) marketplaceUrls.stream().map(urlString -> {
+            try {
+                URL url = new URL(urlString);
+                URLConnection connection = url.openConnection();
+                connection.addRequestProperty("Accept", "application/json");
+                try (Reader reader = new InputStreamReader(connection.getInputStream())) {
+                    Type type = TypeToken.getParameterized(List.class, AddonEntryDTO.class).getType();
+                    return (List<AddonEntryDTO>) Objects.requireNonNull(gson.fromJson(reader, type));
+                }
+            } catch (IOException e) {
+                return List.of();
+            }
+        }).flatMap(List::stream).filter(e -> showUnstable || "stable".equals(((AddonEntryDTO) e).maturity))
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public List<Addon> getAddons(Locale locale) {
+        // TODO: fix that we need to refresh every time
+        refreshSource();
+        return cachedAddons.stream().map(this::fromAddonEntry).collect(Collectors.toList());
+    }
+
+    @Override
+    public Addon getAddon(String id, Locale locale) {
+        String remoteId = id.replace(ADDON_ID_PREFIX, "");
+        return cachedAddons.stream().filter(e -> remoteId.equals(e.id)).map(this::fromAddonEntry).findAny()
+                .orElse(null);
+    }
+
+    @Override
+    public List<AddonType> getTypes(Locale locale) {
+        return new ArrayList<>(TAG_ADDON_TYPE_MAP.values());
+    }
+
+    @Override
+    public void install(String id) {
+        Addon addon = getAddon(id, null);
```

Comment:
The `getAddon(String, Locale)` method may return `null`. We should add an additional check here to prevent NPEs.

Reply:
I also added that in the `CommunityMarketplaceAddonService`